client_web/src/components/Menu.tsx

```diff
@@ -0,0 +1,60 @@
+import { ApolloClient, useMutation } from '@apollo/client';
+import { LOGOUT } from '@graphQL/mutations/userMutations';
+import { UserProfile } from '@utils/types/DataTypes';
+import { NavLink, useNavigate } from 'react-router-dom';
+
+interface Props {
+  userProfile: UserProfile
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  client: ApolloClient<any>
+}
+
+function Menu({ userProfile, client }:Props) {
+  const [logout] = useMutation(LOGOUT);
+  const navigate = useNavigate();
+  const onLogout = async () => {
+    await logout();
+    await client.resetStore();
+    navigate('/auth');
+  };
+
+  return (
+    <div> {userProfile !== undefined && (
+    <div className="m-5 p-5 w-full">
+      {userProfile.role === 1 ? (
+        <div>
```

Comment:
c'est pas tres clair sur ce code là. Tu n'affiche rien si userprofil is undefined ?
Si c'est le cas, ca ne sert a rien de le verifier parce que quelq'un de non connecter n'est pas sensé arriver sur cette page, si ?

Reply:
La sidebar a pour but de différer si elle est utilisée par un opérateur ou un admin. Tu check le rôle, s'il s'agit d'un admin, tu lui affiches les navlink propres aux administrateur et ceux des opérateurs dans le cas contraire.
Pour l'instant le contenu ne justifie pas une telle vérification puisque le routing opérateur reste à faire, mais la structure rendant le composant réutilisable est en place.